### PR TITLE
Fixes #16112 - support for netgroups in LDAP auth source

### PR DESCRIPTION
**Pull request by tstrachota**
Requires following change in ldap_fluff to work:
https://github.com/theforeman/ldap_fluff/pull/57

### CODE REVIEW

**Comment by tstrachota** [comment]
Requires thorough testing. @ares would you mind to take a look?

File: app/controllers/api/v2/auth_source_ldaps_controller.rb
Comment (ares) [inline app/controllers/api/v2/auth_source_ldaps_controller.rb]: why are these strings extracted? they don't seem to be translated

File: app/views/auth_source_ldaps/_form.html.erb
Comment (ares) [inline app/views/auth_source_ldaps/_form.html.erb]: would you mind adding `:label_help => _('By default we map user groups to standard LDAP Group objects. FreeIPA and POSIX LDAP server types supports alternative way of grouping users through Netgroups. Enable this checkbox if using Netgroups is preferred instead of standard groups.')` (feel free to adjust the text)

File: app/views/auth_source_ldaps/_form.html.erb
Comment (ares) [inline app/views/auth_source_ldaps/_form.html.erb]: Would you mind adding `:label_help` with examples of group_base value for supported servers including netgroups? Good examples can be found at https://theforeman.org/manuals/1.15/#4.1.1LDAPAuthentication If you modify that, adding same help to other fields such as base_dn and account would be awesome.

Also updating the manual with netgroups base DN in manual would be appreciated.

**Review by ares — CHANGES_REQUESTED** [review]
Works nicely! We'll need to bump the minimal version of ldap_fluff since it won't work with older version. Feel free to update it when new ldap_fluff gem is released, we can't merge it before that anyway.

Would you mind also adding a hammer PR that would reflect the new attribute for auth source?

**Comment by tstrachota** [comment]
@ares updated

File: app/assets/javascripts/auth_source_ldap.js
Comment (ares) [inline app/assets/javascripts/auth_source_ldap.js]: `.show` would be more explicit I think

File: app/assets/javascripts/auth_source_ldap.js
Comment (ares) [inline app/assets/javascripts/auth_source_ldap.js]: `.hide` would be more explicit I think

File: app/assets/javascripts/auth_source_ldap.js
Comment (ares) [inline app/assets/javascripts/auth_source_ldap.js]: maybe first change the content and then show the field?

File: app/assets/javascripts/auth_source_ldap.js
Comment (ares) [inline app/assets/javascripts/auth_source_ldap.js]: `item` argument is unused

File: app/views/auth_source_ldaps/_form.html.erb
Comment (ares) [inline app/views/auth_source_ldaps/_form.html.erb]: seems like passing `this` is not required

**Review by ares — CHANGES_REQUESTED** [review]
Just few nits, otherwise looks solid and works great.

**Comment by tstrachota** [comment]
@ares updated

**Review by ares — APPROVED** [review]
In my opinion it's ready but I can't merge until the new version of ldap_fluff package is available in nightly repos. Hopefully it gets in soon. Meanwhile I'm setting ready for merge.

Blocked on: https://github.com/theforeman/foreman-packaging/pull/1711

**Comment by ares** [comment]
@tstrachota there some tests broken which might be caused by older base of your branch, please try rebasing, I hope that helps

**Comment by tstrachota** [comment]
@ares rebased

**Comment by ares** [comment]
Good, tests are green now (the failure is unrelated)

Still blocked by https://github.com/theforeman/foreman-packaging/pull/1711

**Comment by tstrachota** [comment]
@ares the packaging PR for ldap_fluff has been merged and PR with docs is ready for review:
https://github.com/theforeman/theforeman.org/pull/921

**Comment by ares** [comment]
Thanks @tstrachota, merged as 88e295d50e85bdad1caeb42638651631a4654bc3. Just for the record, I was waiting because of ldap_fluff 0.4.7 is still not in repos, but that's because of pipeline failures so we won't get new Foreman nightly build before new ldap_fluff lands there either.

**Pull request closed without merging**
